tests/cupyx_tests/scipy_tests/linalg_tests/test_decomp_lu.py

```diff
@@ -32,6 +35,34 @@ def test_lu_factor(self, dtype):
         cupy.testing.assert_allclose(result_cpu[0], result_gpu[0], atol=1e-5)
         cupy.testing.assert_array_equal(result_cpu[1], result_gpu[1])
 
+    @testing.for_float_dtypes(no_float16=True)
+    def test_lu_factor_reconstruction(self, dtype):
+        m, n = self.shape
+        array = cupy.random.randn(m, n, dtype=dtype)
+        lu, piv = cupyx.scipy.linalg.lu_factor(array)
```

Comment:
How about checking the shape of `lu`?
```py
assert lu.shape == (m, n)
assert L.shape == (m, min(m, n))
assert U.shape == (min(m, n), n)
assert len(piv) == min(m, n)
```
It might be enough to check the shape of `lu` for the test, but I think it is better to check also that of `L` and `U` because of readability.